### PR TITLE
Items: parent item

### PR DESCRIPTION
**Pull request by spitfire305**

### CODE REVIEW

File: resources/lang/en/items.php [hunk 1]
```diff
@@ -12,6 +12,8 @@
         'price'     => 'Price',
         'size'      => 'Size',
         'type'      => 'Type',
+        'item_id'   => 'Parent Item ID',
```
Comment (ilestis) [inline resources/lang/en/items.php:15]: Our convention is just "Parent Item" for the parent field, "ID" is a too technical for our users :)

File: resources/lang/en/items.php [hunk 2]
```diff
@@ -26,10 +28,14 @@
         'price'     => 'Price of the item',
         'size'      => 'Size, Weight, Dimensions',
         'type'      => 'Weapon, Potion, Artefact',
+        'parent'    => 'ID of the parent item'
```
Comment (ilestis) [inline resources/lang/en/items.php:31]: Same here, look what is used for the quest's parent field.

File: resources/lang/en/items.php [hunk 3]
```diff
@@ -26,10 +28,14 @@
...
     ],
     'show'          => [
         'tabs'  => [
             'inventories'   => 'Inventories',
         ],
     ],
+    'hints'         => [
+        'items'    => 'A web of interlocking items can be built using the Parent Item field.',
```
Comment (ilestis) [inline resources/lang/en/items.php:39]: I can tell this was copied from quests, where it makes sense. However for items, we can simply use "Organise items by using the parent item field."

File: resources/views/items/form/_entry.blade.php [hunk 4]
```diff
@@ -25,6 +25,14 @@
         @include('cruds.fields.character', ['quickCreator' => true])
     </div>
 </div>
+<div class="row">
```
Comment (ilestis) [inline resources/views/items/form/_entry.blade.php:28]: To be consistent with the rest of the app, the parent field should be placed higher up in the form. please place it after type.

File: app/Models/Item.php [hunk 5]
```diff
@@ -79,6 +85,7 @@ class Item extends MiscModel
         'size',
         'location.name',
         'character.name',
+        'item_id',
     ];
```
Comment (ilestis) [inline app/Models/Item.php:90]: I think `protected $sortable` is missing here for the items subpage. If you have multiple sub items, the "order by" (clicking on the header field name) won't work without it.

File: app/Models/Item.php [hunk 6]
```diff
@@ -182,7 +189,21 @@ public function inventories()
...
     }
-
+    /**
+     * @return \Illuminate\Database\Eloquent\Relations\BelongsTo
+     */
+    public function items()
+    {
+        return $this->hasMany('App\Models\Item', 'item_id', 'id');
+    }
+    /**
```
Comment (ilestis) [inline app/Models/Item.php:199]: Missing empty line

**Review by ilestis — CHANGES_REQUESTED** [review]
Missing from the ticket: we also need the parent item visible in the items/datagrid.blade.php, and being able to order on it :)

File: resources/views/items/datagrid.blade.php [hunk 7]
```diff
@@ -12,6 +12,16 @@
         // Name
         'name',
         'type',
+        //Item_id
+        [
+            'label' => __('items.fields.item_id'),
+            'field' => 'item_id',
+            'render' => function($model) {
+                if ($model->item_id) {                    
+                    return '<a href="' . route('quests.show', $model->item_id) . '">' . e($model->item->name) . '</a>';
```
Comment (ilestis) [inline resources/views/items/datagrid.blade.php:21]: I don't think it should be `quests.show` ;)

File: resources/views/items/datagrid.blade.php [hunk 8]
```diff
@@ -12,6 +12,16 @@
         // Name
         'name',
         'type',
+        //Item_id
+        [
+            'label' => __('items.fields.item_id'),
+            'field' => 'item_id',
+            'render' => function($model) {
+                if ($model->item_id) {
```
Comment (ilestis) [inline resources/views/items/datagrid.blade.php:20]: If you test just on `item_id`, it won't test the permission engine for non-admin viewers of a campaign. You always need to test `model->item`, because if the user doesn't have access to the item, it will return null.

File: resources/views/organisations/datagrid.blade.php [hunk 9]
```diff
@@ -17,10 +17,13 @@
         ],
         // Location
         [
-            'type' => 'avatar',
-            'parent' => 'location',
-            'parent_route' => 'locations',
-            'visible' => $campaignService->enabled('locations'),
+            'label' => __('quests.fields.quest'),
```
Comment (ilestis) [inline resources/views/organisations/datagrid.blade.php:20]: I'm not sure why organisations were changed? Organisations don't have a parent quest

**Review by ilestis — CHANGES_REQUESTED** [review]
Small fixes

File: resources/views/organisations/datagrid.blade.php (hunk 9 above)
Comment (spitfire305) [reply resources/views/organisations/datagrid.blade.php]: It was a mistake, its been reverted.

**Review by ilestis — APPROVED** [review]
